mlflow/utils/environment.py

```diff
@@ -188,6 +191,30 @@ def _is_iterable(x):
         )
 
 
+def infer_pip_requirements(model_uri, flavor, fallback=None):
```

Comment:
Maybe this functions should be moved to `requirements_utils.py`

Reply:
I like this location, since it's a top-level environment utility function. `requirements_utils` are utilities that this method depends on.

Reply:
Agree!